src/lib/colorUtils.ts

```diff
@@ -0,0 +1,61 @@
+import {ColorRgb, Option} from "@/types/common.types";
+
+// RED, GREEN, BLUE represent the percentual share of each
+// color's luminance in the luminance scale of the color spectrum
```

Comment:
for future reference if anyone is curious what happens here and why
```suggestion
// color's luminance in the luminance scale of the color spectrum
// https://stackoverflow.com/questions/9733288/how-to-programmatically-calculate-the-contrast-ratio-between-two-colors
```